### PR TITLE
ContainerBuilder: added operator 'raw' as workaround @strings 

### PR DESCRIPTION
**Pull request by dg**
[Closes #40][Closes nette/nette#1279]

### CODE REVIEW

File: src/DI/ContainerBuilder.php [hunk 1]
```diff
@@ -649,6 +649,10 @@ public function formatStatement(Statement $statement)
 		} elseif ($entity === 'not') { // operator
 			return $this->formatPhp('!?', array($arguments[0]));
 
+		} elseif ($entity === 'raw') { // workaround for string that begins with @
+			return PhpHelpers::formatArgs('?', array($arguments[0]));
+			return $this->formatPhp('!?', array($arguments[0]));
```
Comment (JanTvrdik) [inline src/DI/ContainerBuilder.php]: !

File: src/DI/ContainerBuilder.php [hunk 2]
```diff
@@ -649,6 +649,10 @@ public function formatStatement(Statement $statement)
...
+
```
Comment (xificurk) [inline src/DI/ContainerBuilder.php]: eh? return; return; ?

Comment (dg) [inline src/DI/ContainerBuilder.php]: fixed

**Comment by xificurk** [comment]
Am I overlooking something, or does this mean that I would have to use it in `CompilerExtension` as follows?
```php
$arg = '@string.starting.with.at'
$builder->getDefinition('foo')->setArguments(array('name' => new Statement('raw', array($fooArg))));
```

**Comment by fprochazka** [comment]
This is not very nice.

**Comment by JanTvrdik** [comment]
This is not escaping, this is just slightly better workaround than `trim`.

**Comment by dg** [comment]
Hmm, it will not help when you developing CompilerExtension… I'll use you PR.

That strings like `@service` in arguments are parsed as service references was design flaw. It would be better to use any object, for example  `ServiceReference`:

```php
$arg = '@string.starting.with.at'
$builder->getDefinition('foo')->setArguments(array($fooArg, new ServiceReference('bar')));
```

But even we introduce ServiceReference, we still need to parse `@service` for maintain BC…

**Pull request closed without merging**